common/src/chain/transaction/signature/inputsig/classical_multisig/multisig_partial_signature.rs

```diff
@@ -0,0 +1,439 @@
+// Copyright (c) 2021-2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::chain::{
+    classic_multisig::{ClassicMultisigChallenge, ClassicMultisigChallengeError},
+    ChainConfig,
+};
+
+use super::authorize_classical_multisig::AuthorizedClassicalMultisigSpend;
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct PartiallySignedMultisigChallenge<'a> {
+    /// The signatures that have been added to this challenge. The indices are the indices of the public keys.
+    signatures: &'a AuthorizedClassicalMultisigSpend,
+    /// The message that is being signed.
+    message: &'a [u8],
+}
```

Comment:
The reason I avoided that is because then Eq/PartialEq won't work. I see value in being able to equate this object to another (even though it's not used now).